src/Pim/Bundle/CatalogBundle/tests/integration/ElasticSearch/IndexConfiguration/AbstractPimCatalogIntegration.php

```diff
@@ -62,229 +76,24 @@ private function resetIndex()
     {
-        $params = [];
-        $params['index'] = self::INDEX_NAME;
-        $params['type'] = self::PRODUCT_TYPE;
-
         foreach ($products as $product) {
-            $productBody = [];
-
-            foreach ($product as $field => $value) {
-                $matches = [];
-                if (preg_match('/^(.*)-option$/', $field, $matches)) {
-//                $attributeCode = $matches[1];
-//                $optionParams = $this->currentOptions[$attributeCode][$value];
-//                unset($optionParams['code']);
-//                $productBody[$field] = $optionParams;
-                } elseif (preg_match('/^(.*)-options$/', $field, $matches)) {
-//                $attributeCode = $matches[1];
-//                $options = explode(',', $value);
-//                $optionsParams = [];
-//                foreach ($options as $option) {
-//                    $optionParams = $this->currentOptions[$attributeCode][trim($option)];
-//                    unset($optionParams['code']);
-//                    $optionsParams[] = $optionParams;
-//                }
-//                $productBody[$field] = $optionsParams;
-                } elseif (preg_match('/^(.*)-metric$/', $field, $matches)) {
-                    $productBody[$field] = floatval($value);
-                } elseif (preg_match('/^(.*)-number$/', $field, $matches)) {
-                    $productBody[$field] = floatval($value);
-                } else {
-                    $productBody[$field] = $value;
-                }
-            }
-
-            $params['body'] = $productBody;
-
-            $this->ESClient->index($params);
-        }
-
-        $this->ESClient->indices()->refresh();
-    }
-
-    /**
-     * Prepare a search query with the given clause
-     *
-     * @param array $searchClause
-     * @param array $sortClauses
-     *
-     * @return array
-     */
-    protected function createSearchQuery(array $searchClause, array $sortClauses = [])
-    {
-        $searchQuery = [
-            'index' => self::INDEX_NAME,
-            'type'  => self::PRODUCT_TYPE,
-            'body'  => [],
-        ];
-
-        if (!empty($searchClause)) {
-            $searchQuery['body'] = $searchClause;
-        }
-
-        if (!empty($sortClause)) {
-            $searchQuery['body']['sort'] = $sortClauses;
+            $this->esClient->index(self::INDEX_TYPE, $product['identifier'], $product);
         }
 
-        return $searchQuery;
+        // The indexation in Elasticsearch is an asynchronous process.
+        // And even if this process is Near Real-Time, (see
+        // https://www.elastic.co/guide/en/elasticsearch/guide/master/near-real-time.html#near-real-time)
+        // for more information), we have to wait the indexation is
+        // finished so that our tests are green.
+        // Here, 5s is really more than enough according to the low number
+        // of products we have to index.
+        sleep(5);
```

Comment:
As per the documentation, the segment refresh is done within 1s. So 5s is really too long ;) Or maybe call a _refresh on the index ? It's a blocking action, so once returned, the new documents are guaranteed searcheable. It would make the code less weird and hackish to call for refresh ;)